3D STL Renderer

jvsluis opened the pull request:
Adds the 3d renderer in the create form
Implements resizing of canvas
Fixes some graphical bugs with the upload dropzone sizing. The dropzone now fill the whole containers height correctly

Height of dropzone fixed:
![image](https://github.com/J-Raymer/printlink3d/assets/86749842/4723be9b-e9d0-47aa-92c3-c992095c9342)

3d Renderer:
![image](https://github.com/J-Raymer/printlink3d/assets/86749842/a9000d1f-39b4-4810-8914-c0238e445a64)

jvsluis committed:
add: 3d renderer in create form

Adds the 3d renderer in the create form
Implements resizing of canvas
Fixes some graphical bugs with the upload dropzone sizing. The dropzone now fill the whole containers height correctly

Jasper-Nelligan commented:
Looks like there's merge conflicts that need to be resolved, particularly around the new fileCard component. Should we add the fileCard component at the top of the page above the 3D view?

Another thing I noticed is that if you select a file, click Next, and then go back, there no longer appears to be any file selected.

jvsluis committed:
fix: bug where file didnt appear selected on back

When clicking back on the configure page after selecting an STL file, the file would fail to re-render. This fix ensures the the renderer is recreated from the previously selected file.

jvsluis committed:
remove: selected state no longer required

jvsluis commented:
With the 3D Renderer I dont see a point to the file card. Maybe I'm wrong but the card would add no information. The file title, and the change button are sufficient to replace the file. Let me know what you think.

Also - I fixed the back button bug. Thx for spotting that.

Jasper-Nelligan commented:
> With the 3D Renderer I dont see a point to the file card. Maybe I'm wrong but the card would add no information. The file title, and the change button are sufficient to replace the file. Let me know what you think.
> 
> Also - I fixed the back button bug. Thx for spotting that.

I guess it's not necessary anymore, but it just seems a shame to waste the fileCard. Oh well. I think you're good to remove it

Jasper-Nelligan commented on app/src/routes/upload.js:
```diff
@@ -1,44 +1,172 @@
+      }
+
+      if (containerRef.current !== null && sceneState.children.length > 3) {
+        console.debug("added renderer");
+        // Init the renderer object
+        // Add the dom object the the containerRef so we can add it via react
+        renderer.setSize(800, 600);
+        camera.aspect = 800 / 600;
+        camera.updateProjectionMatrix();
+
+        window.addEventListener('resize', onWindowResize, false);
+        resize3DViewer();
+
+        containerRef.current.appendChild(renderer.domElement);
+        draw();
+        return () => {
+          console.debug("attempting to remove renderer");
+          window.removeEventListener('resize', onWindowResize, false);
+          if (containerRef.current !== null && containerRef.current.hasChildNodes()) {
+            console.debug("removed renderer");
+            containerRef.current.removeChild(renderer.domElement);
+            updateFile(null);
+          }
+        }
+      }
```
Are the console logs necessary still?

jvsluis committed:
merge: merged in main, resolved conflicts

Also removed debugger logging and fixed the button positions
